include/rmm/detail/error.hpp

```diff
@@ -205,5 +207,25 @@ class out_of_range : public std::out_of_range {
                 << cudaGetErrorString(status__) << std::endl;                   \
     }                                                                           \
     assert(status__ == cudaSuccess);                                            \
-  } while (0);
+  } while (0)
+#endif
+
+/**
+ * @brief Assertion that logs a CRITICAL log message on failure.
+ */
+#ifdef NDEBUG
+#define RMM_LOGGING_ASSERT(_expr) (void)0
```

Comment:
I've never thought to ask... is it bad practice to assert in debug builds?

Reply:
Confusing question. Assert should be a no-op in non-debug builds, which is what this does.